client.go

```diff
@@ -27,10 +29,24 @@ var (
 	ErrBadRequest = errors.New("error: bad request")
 )
 
+// Error API Response. Contains the error message as well as the type of error
+type Error struct {
+	Message string `json:"message"`
+	Type    string `json:"type"`
+}
+
 //URLBuilder is the interface for building URLs
 //go:generate mockery --name URLBuilder
 type URLBuilder interface {
 	SearchDocumentURL(vaultID string) string
+	GetUserURL(userId []string) string
```

Comment:
I'm not 100% sure how I feel about this pattern. The client needs to know so much about other packages. I wonder if it would be easier if we just mocked the `Do` function in our tests then we can move the url stuff to the appropriate packages

Reply:
IMO this is fine. The client only needs to know about the URLBuilder interface, not its implementation. The nice thing about this URL builders is we can mock the response using the HTTP test and actually use the http client on tests
https://github.com/FirstVisit/go-truevault/blob/7bde5dcc6931abccc1b2933c81f4d0554f42afa4/document/document_test.go#L66-L78
I guess the reason why it feels like the client is doing so much is that this interface is in client.go. We can move this to a separate file if that helps